src/pages/admin/dashboard.vue

```diff
@@ -0,0 +1,107 @@
+<script setup lang="ts">
+import { fetcher } from "@/models/api";
+import { useAxios } from "@vueuse/integrations/useAxios";
+import { useI18n } from "vue-i18n";
+import { ROLE } from "@/constants";
+import { useTitle } from "@vueuse/core";
+import { computed } from "vue";
+
+useTitle("Admin - Dashboard | Normal OJ");
+const { t } = useI18n();
+const {
+  data: userSummary,
+  error: userError,
+  isLoading: userLoading,
+} = useAxios<UserSummary>("/user/summary", fetcher);
+const {
+  data: courseSummary,
+  error: courseError,
+  isLoading: courseLoading,
+} = useAxios<CourseSummary>("/course/summary", fetcher);
+const courseSummaryTotal = computed(() => {
+  if (!courseSummary.value) return null;
+  return {
+    userCount: courseSummary.value.breakdown.reduce((acc, cur) => acc + cur.userCount, 0),
+    problemCount: courseSummary.value.breakdown.reduce((acc, cur) => acc + cur.problemCount, 0),
+    submissionCount: courseSummary.value.breakdown.reduce((acc, cur) => acc + cur.submissionCount, 0),
+    homeworkCount: courseSummary.value.breakdown.reduce((acc, cur) => acc + cur.homeworkCount, 0),
+  };
+});
+</script>
+
+<template>
+  <div class="mb-4 flex flex-col gap-y-4">
+    <h1 class="text-xl font-bold">
+      {{ t("admin.dashboard.usercount.title") + ` (${userSummary?.userCount || "-"})` }}
+    </h1>
+    <data-status-wrapper :error="userError" :is-loading="userLoading">
+      <template #loading>
+        <skeleton-table :col="2" :row="5" />
+      </template>
+      <template #data>
+        <table class="table table-compact w-full">
+          <thead>
+            <tr>
+              <th>#</th>
+              <th>{{ t("admin.dashboard.usercount.count") }}</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr v-for="{ role, count } in userSummary?.breakdown" :key="role">
```

Comment:
Although we can use `v-for` to loop over `undefined`, I think it does not make sense. Maybe we can change it to 

```suggestion
            <tr v-for="{ role, count } in userSummary!.breakdown" :key="role">
```

Reply:
yeah you're right, I almost forgot we have "!" could be used. Let me fix this.

All of them in side <template #data> in <data-status-wrapper /> could use exclamation.

Reply:
Maybe out of the scope of this PR, but maybe we can use slot props to ensure type consistency, i.e. no more `!` needed.
Anyway, LGTM now 👍

Reply:
will create an issue for this, thanks